src/Core/Scheduler/Providers/PointCloudProvider.js

```diff
@@ -160,6 +161,7 @@ export default {
         layer.update = PointCloudProcessing.update;
         layer.postUpdate = PointCloudProcessing.postUpdate;
 
+        layer.pickObjectsAt = (view, mouse) => Picking.pickPointsAt(view, mouse, layer);
```

Comment:
`pickObjectsAt` must be moved in geometry layer

Reply:
(small aside: please avoid this wording. "You must do this" is kinda rude)

There's no perfect place for this kind of code actually. PR #546 aimed to fix this but it couldn't be merged yet...

Here you're nitpicking on one line and your argument "Provider is to provide things" can be used to not move the picking code in `Layer.js` : "Layer should only contain generic thins but Picking.pickPointsAt is a custom pick code".

So I suggest we move and work on improving #546 (or submit another PR that would solve this problem).

Reply:
It can have several providers return the same type of object, THREE.Object3D for example, it makes more sense to move it in the construction of the geometric layer than to put picking code in  each provider that returns an object3D

Reply:
> put  picking code in each provider that returns an object3D

- I'm doing this solely for the PointCloudProvider
- this line will go away as soon as #546 is merged
- I disagree that putting this pointcloud specific line is better in GeometryLayer constructor.

Will you block this PR for this?

Reply:
Are you ok with this [comment](https://github.com/iTowns/itowns/pull/635#discussion_r166295422)?

Otherwise, I agree, it makes more sense, first of all, to work and merge #546

Reply:
No.
I'd prefer to first merge this one, and then merge the more complicated one.

Reply:
Reminder: this PR doesn't remove nor add feature. It simply groups the picking-from-window-coordinate in the same module. I don't even see why we're arguing so much about it :(

Reply:
add a temporary comment : to indicate that it'll not stay in the provider